src/ssh-generator/ssh-generator.c

```diff
@@ -103,14 +106,19 @@ static int make_sshd_template_unit(
                 if (r < 0)
                         return r;
 
+                /* only change AuthorizedKeysFile if an ephemeral key is present */
+                if (access("/run/ssh.authorized_keys.root", F_OK) >= 0)
+                        authorized_keys_file = " -o 'AuthorizedKeysFile /run/ssh.authorized_keys.root .ssh/authorized_keys'";
```

Comment:
hmm, so is this authorized_keys file for root or for all users? it has "root" in the name, but as i understand the setting it will apply to all users.

Reply:
Yes this will apply for all users.

IMO this isn't necessarily an issue as root can already become other users via su etc.

Reply:
I would be keen to change the credential name tbh, I would prefer the name to indicate that it is not persistent.

Reply:
yeah, please use a different credential name that emphasizes that this is only going to be used transiently, and drop the reference to "root"

Reply:
My working name was ssh.ephemeral-key.all-users